tests/test_kms.py

```diff
@@ -60,19 +60,19 @@ def test_kms_remove_matched(self):
                 "Version": "2008-10-17",
                 "Statement": [
                     {
-                      "Sid": "DefaultRoot",
-                      "Effect": "Allow",
-                      "Principal": {
-                        "AWS": "arn:aws:iam::123456789012:root"
-                      },
-                      "Action": "kms:*",
-                      "Resource": "*"
+                        "Sid": "DefaultRoot",
+                        "Effect": "Allow",
+                        "Principal": {
+                            "AWS": "arn:aws:iam::644160558196:user/scot@sixfeetup.com"
```

Comment:
Is it acceptable to put this user in here?  It was originally using a fake root account, but that was causing the error.  I tried with user `kapil` but using an admin account produces a different error.

Reply:
I'd prefer to avoid the email addresses for recorded users, we setup a separate iam account to avoid having extraneous users in recorded tests afaicr.

Reply:
understood - I figured it wasn't ideal - that's why I asked.  

The error that is happening when running functional tests (or if trying to re-record the test) is that it no longer likes the bogus 123456789012:root account that was used in the tests: 

Here is the error I get if I try to use an admin account:

```
Traceback (most recent call last):
  File "test_kms.py", line 86, in test_kms_remove_matched
    "kms:Put*"
...
...
MalformedPolicyDocumentException: An error occurred (MalformedPolicyDocumentException) when calling the PutKeyPolicy operation: The new key policy will not allow you to update the key policy in the future.
```

> we setup a separate iam account to avoid having extraneous users in recorded tests afaicr.

Are you saying there is a specific account I am supposed to use?  I didn't figure it out by looking at the other tests.  There is a smattering of different users throughout them.

Reply:
@scotwk I think maybe he's referring to the `custodian-mu` account?

Reply:
Whether using `user/scot@sixfeetup.com` or `role/custodian-mu`, I get a `MalformedPolicyDocumentException`:

> An error occurred (MalformedPolicyDocumentException) when calling the PutKeyPolicy operation: The new key policy will not allow you to update the key policy in the future.

Are we perhaps going to have to run the functional tests using credentials for a certain user?

Reply:
I think we can drop the user entirely in the `named` case because it's not necessary for what we're trying to test. For the `matched` case we should be able to use `role/custodian-mu`, but the confusing thing there is the test assertion: aren't we trying to match and remove `DefaultRoot` and `SpecificAllow`? Exploring in 3ebea8b415c955304ea8a10d5391b92bd7a2ef6f ...